### ISSUE TITLE
BasicObjectsRoundTripAndMatch fails

### ISSUE THREAD

**Issue by lewing** [dotnet/runtime#107629]
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=803627
Build error leg or test failing: System.Resources.Extensions.Tests.FormattedObject.BasicObjectTests.BasicObjectsRoundTripAndMatch
Pull request: https://github.com/dotnet/runtime.git/pull/107610

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

```json
{
  "ErrorMessage": "BasicObjectsRoundTripAndMatch",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=803627
**Error message validated:** `[BasicObjectsRoundTripAndMatch`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 9/10/2024 4:55:27 PM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[806061](https://dev.azure.com/dnceng-public/public/_build/results?buildId=806061)|dotnet/runtime|[System.Resources.Extensions.BinaryFormat.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=806061&view=ms.vss-test-web.build-test-results-tab&runId=20832774&resultId=164241)|dotnet/runtime#107617|
|[805378](https://dev.azure.com/dnceng-public/public/_build/results?buildId=805378)|dotnet/runtime|[System.Resources.Extensions.BinaryFormat.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=805378&view=ms.vss-test-web.build-test-results-tab&runId=20815080&resultId=173572)|dotnet/runtime#107617|
|[803627](https://dev.azure.com/dnceng-public/public/_build/results?buildId=803627)|dotnet/runtime|[System.Resources.Extensions.Tests.FormattedObject.BasicObjectTests.BasicObjectsRoundTripAndMatch](https://dev.azure.com/dnceng-public/public/_build/results?buildId=803627&view=ms.vss-test-web.build-test-results-tab&runId=20757462&resultId=136787)|dotnet/runtime#107610|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|2|3|

**Comment by ericstj** [comment]
This is hitting an assert in the JIT https://github.com/dotnet/runtime/blob/f722e5c5b72000a06ce267a9bee222a8c3be4696/src/coreclr/jit/gcencode.cpp#L3891

 Seems like something we should look at.  Dumps are available in the failing tests.
https://dev.azure.com/dnceng-public/public/_build/results?buildId=805378&view=ms.vss-test-web.build-test-results-tab&runId=20815080&resultId=173572&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab
https://dev.azure.com/dnceng-public/public/_build/results?buildId=806061&view=ms.vss-test-web.build-test-results-tab&runId=20832774&resultId=164241&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab

**Comment by JulieLeeMSFT** [comment]
@AndyAyersMS, PTAL.

**Comment by AndyAyersMS** [comment]
The first two failures with the jit assert ([806061](https://dev.azure.com/dnceng-public/public/_build/results?buildId=806061) and [805378](https://dev.azure.com/dnceng-public/public/_build/results?buildId=805378) are from a post-9.0 PR #107766 and quite likely related to the change the PR is making.

The third failure is something different and (at least at this stage) is not manifestly a codegen issue.
```
    System.Resources.Extensions.Tests.FormattedObject.BasicObjectTests.BasicObjectsRoundTripAndMatch(value: WeakReference`1 { }, _: [BinaryFormatTests.TypeSerializableValue, BinaryFormatTests.TypeSerializableValue]) [FAIL]
      Assert.Equal() Failure: Values differ
      Expected: 479
      Actual:   407
      Stack Trace:
        /_/src/libraries/System.Resources.Extensions/tests/BinaryFormatTests/FormattedObject/BasicObjectTests.cs(40,0): at System.Resources.Extensions.Tests.FormattedObject.BasicObjectTests.BasicObjectsRoundTripAndMatch(Object value, TypeSerializableValue[] _)
           at InvokeStub_BasicObjectTests.BasicObjectsRoundTripAndMatch(Object, Span`1)
        /_/src/libraries/System.Private.CoreLib/
```
@ericstj can you suggest new area and owner?

**Comment by ericstj** [comment]
@adamsitnik  Seems like a single hit of `BasicObjectsRoundTripAndMatch` failing.

The input data is `value: WeakReference`1 { }, _: [BinaryFormatTests.TypeSerializableValue, BinaryFormatTests.TypeSerializableValue]`

I noticed you made this change: https://github.com/dotnet/runtime/issues/104905 in main,  and are backporting it to release/9.0 in https://github.com/dotnet/runtime/pull/107540.

I think that PR would fix this issue.

**Comment by ericstj** [comment]
Resolving as the fix was backported in #107540
